src/main/java/appeng/client/gui/me/search/SearchPredicates.java

```diff
@@ -1,84 +1,51 @@
 package appeng.client.gui.me.search;
 
-import java.util.Objects;
+import java.util.LinkedList;
+import java.util.List;
 import java.util.function.Predicate;
-import java.util.regex.Pattern;
-import java.util.regex.PatternSyntaxException;
 
-import appeng.core.AEConfig;
 import appeng.menu.me.common.GridInventoryEntry;
-import appeng.util.Platform;
 
 final class SearchPredicates {
 
-    static Predicate<GridInventoryEntry> fromString(String searchString, RepoSearch repoSearch) {
-        if (searchString.startsWith("@")) {
-            return createModIdPredicate(searchString.substring(1))
-                    .or(createModNamePredicate(searchString.substring(1)));
-        } else if (searchString.startsWith("*")) {
-            return createIdPredicate(searchString.substring(1));
-        } else if (searchString.startsWith("#")) {
-            var pattern = createPattern(searchString.substring(1));
-            return new TagPredicate(pattern);
-        } else {
-            var pattern = createPattern(searchString);
+    static Predicate<GridInventoryEntry> fromString(String searchString) {
+        List<Predicate<GridInventoryEntry>> gridPredictions;
 
-            if (AEConfig.instance().isSearchTooltips()) {
-                // The tooltip obviously includes the display name too
-                return createTooltipPredicate(pattern, repoSearch);
-            } else {
-                return createNamePredicate(pattern);
-            }
-        }
-    }
+        String[] orParts = searchString.split("\\|");
 
-    private static Predicate<GridInventoryEntry> createModIdPredicate(String searchText) {
-        var searchPattern = createPattern(searchText);
-        return entry -> {
-            var what = Objects.requireNonNull(entry.getWhat());
-            return searchPattern.matcher(what.getModId()).find();
-        };
-    }
-
-    private static Predicate<GridInventoryEntry> createModNamePredicate(String searchText) {
-        var searchPattern = createPattern(searchText);
-        return entry -> {
-            var what = Objects.requireNonNull(entry.getWhat());
-            return searchPattern.matcher(Platform.getModName(what.getModId())).find();
-        };
-    }
+        if (orParts.length == 1) {
+            gridPredictions = getPredicates(searchString);
```

Comment:
```suggestion
            gridPredictions = getPredicates(orParts[0]);
```

Otherwise you have an edge case with strings which end in `|` and contain no other pipe,
these search strings currently find no items. 

e.g. `redstone` finds redstone dust etc., but `redstone|` won't find anything

Reply:
Thanks, tested and pushed through a fix :)